app/Console/Commands/SyncBDclub.php

```diff
@@ -0,0 +1,220 @@
+<?php
+
+namespace Lara\Console\Commands;
+
+use Carbon\Carbon;
+use ICal\Event;
+use ICal\ICal;
+use Illuminate\Console\Command;
+use it\thecsea\simple_caldav_client\SimpleCalDAVClient;
+use Lara\Club;
+use Lara\ClubEvent;
+use Lara\Person;
+use Lara\Schedule;
+use Lara\Section;
+use Lara\Shift;
+
+/**
+ * To work with this, you need a file named bd_credentials.php in the config folder
+ * content:
+ * <code>
+ * return [
+ * 'host'=>
+ * 'example.com',
+ * 'user' => 'foo',
+ * 'password' => 'bar',
+ * 'calendarName' =>'foobar'
+ * ];
+ * </code>
+ */
+class SyncBDclub extends Command
+{
+    const BD_TEMPLATE_NAME = 'BD Template';
+    
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'lara:sync-bd-club';
+    
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'download all events from bd club';
+    
+    const DATE_TIME_FORMAT_PREFIX = 'Ymd';
+    
+    const DATE_TIME_FORMAT_SUFFIX = 'His';
+    
+    const BD_SECTION_NAME = 'BD-Club';
+    
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        parent::__construct();
+    }
+    
+    /**
+     * Execute the console command.
+     *
+     * @return mixed
+     */
+    public function handle()
+    {
+        //
+        
+        $section = Section::where('title', '=', self::BD_SECTION_NAME)->first();
+        if (is_null($section)) {
+            $section = new Section();
+            $section->title = self::BD_SECTION_NAME;
+            $section->section_uid = hash("sha512", uniqid());
+            $section->save();
+            $club = new Club();
+            $club->clb_title = $section->title;
+            $club->save();
+        }
+        
+        
+        $caldavClient = new SimpleCalDAVClient();
+        
+        
+        $caldavClient->connect(config('bd_credentials.host'),
+            config('bd_credentials.user'), config('bd_credentials.password'));
+        $arrayOfCalenars = $caldavClient->findCalendars();
+        $caldavClient->setCalendar($arrayOfCalenars[config('bd_credentials.calendarName')]);
+        // $this->info(var_dump($arrayOfCalenars));
+        $gmt = new \DateTimeZone('GMT');
+        $year = strftime('%Y');
+        $month = strftime('%m');
+        
+        $from = new \DateTime($year.'-'.$month.'-01');
+        $until = new \DateTime($from->format('Y-m-d'));
+        $from->modify('previous month');
+        $until->modify('next month')->modify('-1 day');
+        $from->setTimezone($gmt);
+        $until->setTimezone($gmt);
+        
+        //$this->info($this->formatDateTime($from));
+        $events = $caldavClient->getEvents($this->formatDateTime($from), $this->formatDateTime($until));
+        /* @var $template Schedule */
```

Comment:
Whats up with this comment? Seems too verbose to me

Reply:
this is for autocompletion in the ide
without the ide marks every attribute as magic attribute because it does not know what kind of object is it